embedfs/dir_test.go

```diff
@@ -0,0 +1,63 @@
+package embedfs
+
+import (
+	"os"
+	"testing"
+
+	"github.com/go-git/go-billy/v6/embedfs/internal/testdata"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+// Directory operation tests adapted from test/dir_test.go for read-only embedfs
+
+func TestDir_ReadDirNested(t *testing.T) {
+	t.Parallel()
+
+	fs := New(testdata.GetTestData())
+
+	// Test reading nested directories
+	entries, err := fs.ReadDir("/testdata/subdir")
+	require.NoError(t, err)
+	assert.Len(t, entries, 1)
+	assert.Equal(t, "nested.txt", entries[0].Name())
+	assert.False(t, entries[0].IsDir())
+}
+
+func TestDir_ReadDirFileInfo(t *testing.T) {
+	t.Parallel()
+
+	fs := New(testdata.GetTestData())
+
+	entries, err := fs.ReadDir("/testdata")
+	require.NoError(t, err)
+	
+	// Verify all entries have proper FileInfo
+	for _, entry := range entries {
+		assert.NotEmpty(t, entry.Name())
+		assert.NotNil(t, entry.ModTime())
+		assert.Greater(t, entry.Size(), int64(-1)) // Size can be 0 but not negative
+	}
+}
```

Comment:
Instead of having its own test, this could be an improvement for `TestReadDir` - adding expectations around `FileInfo` and verifying them.